### PR TITLE
Manifest Slice C1: polling-range manifest (#293)

### PR DESCRIPTION
**Pull request by dewet22**
## Summary

Third of four planned manifest slices (A semantics-routing+renames [merged, #365] → B fact-table consolidation [merged, #366] → **C1 polling-range manifest** → C2 detect()/plant.py parity unification [deferred, separate design] → D write surface), all landing together as **2.10.0**.

A pure refactor — zero intended wire-behaviour change — replacing hardcoded, capability-gated register ranges in `Client.load_config()` and `_refresh_banks()` with lookups into two new declarative tables in `manifest.py`:

- **New:** `manifest.RegisterRange` (a small frozen dataclass), `manifest.LOAD_CONFIG_RANGES` (5 keys) + a standalone `LOAD_CONFIG_THREE_PHASE_RANGES`, `manifest.REFRESH_IR_RANGES` (3 keys), and a generic `gated_ranges(table, model, arm_fw=None)` accessor.
- **Closes a Slice B gap:** `is_ems`/`is_gateway` were never migrated onto `manifest.CAPABILITIES` in the prior slice — that slice's own final review flagged it for follow-up. Both are now proper `CAPABILITIES` entries, and `PlantCapabilities`'s corresponding properties consult them, matching the other seven predicates already migrated.
- `is_three_phase` and `has_extended_slots` stay as literal inline checks in `load_config()`, deliberately — their relative order is a real behaviour contract (two real models, `ALL_IN_ONE_HYBRID` and `HYBRID_HV_GEN3`, have both facts true simultaneously, and the three-phase HR(1000+) block must poll before HR(240)). Folding either into the generic table would have silently reordered requests for those two models.

## A worthwhile mid-implementation discovery

The plan's literal code for `load_config()`'s tail facts called `manifest.gated_ranges()` directly against the manifest table — but that bypasses `PlantCapabilities`'s own properties, which broke two pre-existing tests that use `unittest.mock.PropertyMock` to patch `has_hv_cabinet_block`/`has_peak_shaving_block` (both facts have empty `CAPABILITIES` sets today — no real hardware has confirmed either block yet — so mocking the property is the only way to test the "when eventually confirmed" branch). Fixed by gating via `getattr(caps, name)` instead, which is behaviourally identical for every real model (each property just calls `manifest.has_capability(...)` internally) while still honouring the test seam. `_refresh_banks()`'s three facts (`is_three_phase`/`is_ems`/`is_gateway`) all have real, non-empty model membership, so they use the plan's original `gated_ranges()` call as designed.

## Test plan

- [x] Direct tests for `RegisterRange`, both tables' exact values, and `gated_ranges()` (including the `Model | None` edge case)
- [x] Every existing `load_config()`/`refresh()` test passes unmodified — the regression fence for wire-request order
- [x] New regression test pinning the `HYBRID_HV_GEN3` both-true ordering contract (HR(1000+) before HR(240)) — passed on first run against the implementation, confirming the order-preservation claim holds by construction, not luck
- [x] Full suite (1534 tests), mypy, ruff, and tox all green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Improved device compatibility so configuration and refresh requests are built dynamically for more inverter types, including three-phase, EMS, and gateway models.
  * Expanded support for additional device capability checks, helping the app handle a wider range of hardware accurately.

* **Bug Fixes**
  * Corrected the order of some configuration reads to ensure important device data is requested in the expected sequence.

* **Tests**
  * Added coverage for new capability handling and request ordering.

### CODE REVIEW

File: givenergy_modbus/client/client.py [hunk 1]
```diff
@@ -333,20 +334,27 @@ def _strategise(
...
 
+def _request_for_range(r: manifest.RegisterRange, device_address: int) -> TransparentRequest:
+    """Build the read request for one RegisterRange at device_address."""
+    if r.reg_type == "HR":
+        return ReadHoldingRegistersRequest(
+            base_register=r.base_register, register_count=r.register_count, device_address=device_address
+        )
+    return ReadInputRegistersRequest(
+        base_register=r.base_register, register_count=r.register_count, device_address=device_address
+    )
```
[the remark this answers is not in the document]
Comment (dewet22) [reply givenergy_modbus/client/client.py]: Good catch, thanks — fixed in c771f5a: `_request_for_range` now raises `ValueError` for an unrecognised `reg_type` instead of silently falling through to an IR request.

File: givenergy_modbus/model/manifest.py [hunk 2]
```diff
@@ -3,14 +3,18 @@
...
+(`LOAD_CONFIG_RANGES`, `REFRESH_IR_RANGES`, `gated_ranges`); C2/D will add
+detect-time polling and the write surface.
 
 Keys are the RESOLVED specific ``Model`` (from ``resolve_model``), never the coarse
 family. Every accessor takes ``arm_fw`` so a firmware column exists in the contract
 from day one; no Slice A row uses it (future users: slot_map's ``HYBRID_GEN3 and
 arm_fw > 302`` extended-slot wrinkle, the Gateway V1/V2 layout variant — Slice B+).
 """
 
+from dataclasses import dataclass
```
[the remark this answers is not in the document]
Comment (dewet22) [reply givenergy_modbus/model/manifest.py:16]: Declining this one — RegisterRange's sibling type, ProbeRange (client.py), already uses a plain `reg_type: str` with the same `# "HR" or "IR"` comment convention, and I've deliberately kept the two types unreconciled in this slice (that unification is scoped to a later slice). Adding Literal only to RegisterRange would create an inconsistency rather than close one. I did take the companion suggestion in the other thread — _request_for_range now raises on an unrecognised value, which covers the correctness concern without the type-level change.

File: givenergy_modbus/model/manifest.py [hunk 3]
```diff
@@ -183,3 +191,72 @@ def has_extended_slots(model: Model, arm_fw: int | None = None) -> bool:
         return True
     threshold = _EXTENDED_SLOT_FIRMWARE_GATE.get(model)
     return threshold is not None and arm_fw is not None and arm_fw > threshold
+
+
+@dataclass(frozen=True)
+class RegisterRange:
+    """A register range to poll: type, base address, and count (#293 Slice C1)."""
+
+    reg_type: str  # "HR" | "IR"
```
[the remark this answers is not in the document]
Comment (dewet22) [reply givenergy_modbus/model/manifest.py:200]: Same reasoning as the sibling thread on the import line — declining in favour of matching ProbeRange's existing plain-str convention; see that reply for the full rationale.

**Comment by dewet22-codex** [comment]
I have started a review pass on this PR.

**Review by dewet22-codex — APPROVED** [review]
Reviewed c771f5a. I don’t see any blocking concerns: the manifest range tables preserve the previous load_config()/refresh request ordering in the paths I checked, EMS/Gateway capability migration stays consistent with the old predicates, and the unsupported RegisterRange.reg_type path now fails loudly rather than falling through to IR. Focused local checks passed: `uv run --group test pytest tests/model/test_manifest.py tests/model/test_plant.py tests/client/test_load_refresh.py tests/client/test_detect.py tests/model/test_fixture_golden_master.py` (316 passed).
